### PR TITLE
refactor(controller): add abstractclass; move publish trigger into common.release

### PR DESCRIPTION
**Pull request by wklken**
### Description

- publish trigger挪到 common.release下
- controller 所有涉及模板方法的父类增加 abc约束
- 去除 attrs, 使用dataclass替代
- 去掉 gateway.is_micro_gateway
- 重构 common/event
- 重构 common/release/publish.py

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [x] PR 中包含单元测试 (include unit test)
- [x] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Comment by wklken** [comment]
需要补充单测的:
1. common/release
2. common/event

File: src/dashboard/apigateway/apigateway/biz/release.py [hunk 1]
```diff
@@ -86,30 +62,38 @@ def batch_get_stage_release_status(stage_ids: List[int]):
...
-        release_event_map = ReleaseHandler.get_latest_publish_event_by_release_history_ids(latest_release_history_ids)
+        publish_id_to_latest_event_map = ReleaseHandler.get_publish_id_to_latest_publish_event_map(
+            latest_release_history_ids
+        )
 
         # 遍历结果集
         stage_publish_status = {}
         for release_history in latest_release_histories:
-            stage_publish_status[release_history.stage_id] = {"publish_id": release_history.id}
+            stage_id = release_history.stage.id
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/biz/release.py:75]: release_history.stage_id 少一次 db 查询？

File: src/dashboard/apigateway/apigateway/common/release/publish.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+logger = logging.getLogger(__name__)
+
+
+def _is_gateway_ok_for_releasing(release: Release) -> Tuple[bool, str]:
+    """网关发布校验"""
+    if not release:
+        return False, "release is None, ignored"
+
+    gateway_id = release.gateway.pk
+    # 剔除非微网关托管的网关
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/common/release/publish.py:44]: 剔除停用的网关

File: src/dashboard/apigateway/apigateway/common/release/publish.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        delay_on_commit(
+            rolling_update_release,
+            gateway_id=release.gateway_id,
+            publish_id=publish_id,
+            release_id=release.pk,
+        )
+    return None
+
+
+def _trigger_revoke_disable_publish(
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/common/release/publish.py:114]: _trigger_revoke_disable_publish  => _trigger_revoke_publish_by_disable

_trigger_revoke_delete_publish => _trigger_revoke_publish_by_delete

remove_disable_publish 看方法名不知道这个的具体含义；看下面调用，认为是revoke发布，因为网关、环境被禁用了

**Review by alex-smile — APPROVED** [review]

File: src/dashboard/apigateway/apigateway/biz/release.py (hunk 1 above)
Comment (wklken) [reply src/dashboard/apigateway/apigateway/biz/release.py:75]: done

File: src/dashboard/apigateway/apigateway/common/release/publish.py (hunk 2 above)
Comment (wklken) [reply src/dashboard/apigateway/apigateway/common/release/publish.py:44]: done

File: src/dashboard/apigateway/apigateway/common/release/publish.py (hunk 3 above)
Comment (wklken) [reply src/dashboard/apigateway/apigateway/common/release/publish.py:114]: _trigger_revoke_publish_for_disable
_trigger_revoke_publish_for_deleting

**Pull request merged**